### PR TITLE
Add hex alpha channel support and require Node.js 6

### PR DESCRIPTION
**Pull request by Mottie**
* Added `xo`.
* Updated `ava`.
* Updated node version.
* Add support for 4 and 8 digit hex values.
* Added tests

```js
hexRgb('#4183c488');
//=> [65, 131, 196, 136]

hexRgb('#0008');
//=> [0, 0, 0, 136]
```

---

I would also like to add support to output the RGB value in an object. For example:

```js
hexRgb('#4183c488', {output: 'object'});
//=> {R: 65, G: 131, B: 196, A: 136}
```

This should be added in a new PR, but I wanted to get your thoughts on it.

### CODE REVIEW

File: test.js
Comment (sindresorhus) [inline test.js]: Use `t.deepEqual` instead of joining into a string.

File: index.js
Comment (sindresorhus) [inline index.js]: Use `slice` instead of `substring`. See: https://github.com/sindresorhus/eslint-plugin-unicorn/issues/65

File: index.js
Comment (sindresorhus) [inline index.js]: For readability reasons, can you make these two regexes functions with descriptive names?

Comment (sindresorhus) [reply index.js]: And `0-9` => `\d`

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
Nice work! Thanks for the high-quality PR :)

**Comment by sindresorhus** [comment]
> hexRgb('#4183c488', {output: 'object'});
> //=> {R: 65, G: 131, B: 196, A: 136}

I would prefer:

```js
hexRgb('#4183c488', {format: 'object'});
//=> {red: 65, green: 131, blue: 196, alpha: 136}
```

**Comment by Mottie** [comment]
All done! 😸

**Comment by Mottie** [comment]
I didn't see a CLI repo for this, otherwise I would have updated it as well.

File: index.js
Comment (sindresorhus) [inline index.js]: The `format` option should be a separate PR. And I think it should be the new default format.

File: index.js
Comment (sindresorhus) [inline index.js]: Alpha should always be included, for consistency.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.js
Comment (Mottie) [reply index.js]: Do you think always including an alpha value would break existing use cases?

File: index.js
Comment (Mottie) [reply index.js]: Ok, I'll wait until this PR has been merged.

File: index.js
Comment (sindresorhus) [reply index.js]: It would, but this will have to be a major version anyways, as we're targeting a newer Node.js version, so it's fine. If anyone really needs the existing behavior, we can consider adding an option when/if they complain.

File: index.js
Comment (sindresorhus) [inline index.js]: Better to fix the violation than ignoring: https://eslint.org/docs/rules/no-mixed-operators

**Review by sindresorhus — APPROVED** [review]

**Comment by sindresorhus** [comment]
Great! Thanks for your patience :)

**Pull request merged**
